### PR TITLE
produce better error codes

### PR DESCRIPTION
**Pull request by pkit**
if one of the queries in clustered environment failed, the final response was still "200 OK"
right now the server will choose randomly one of the error codes from clients and return them to user instead of "OK"
correct error messages can still be parsed from `X-Nexe-Error` response header

### CODE REVIEW

**Comment by mgeisler** [comment]
> if one of the queries in clustered environment failed, the final response was still "200 OK"
> right now the server will choose randomly one of the error codes from clients and return them to user instead of "OK"
> correct error messages can still be parsed from `X-Nexe-Error` response header

Could you rephrase the commit message a little? The "right now" part confused me when I read it. I understand now that what you really meant was:

> Before, if one of the queries in clustered environment failed, the final response was "200 OK".
> 
> Now the server will correctly propagate a non-2xx error code to the client. The error returned is picked at random among the individual errors returned from the cluster.
> 
> The individual error messages can still be parsed from the `X-Nexe-Error` response header.

In writing this I found it quite difficult to explain the random behavior. So perhaps this is a good indicator that we should not pick an error code at random. I suggest we use

```
final_response.status = max(final_response.status, conn.resp.status)
```

instead and explain that we return the maximum status code. HTTP status codes happens to be ordered roughly by severity with 4xx being client errors and 5xx being server errors.

**Comment by pkit** [comment]
We can choose the biggest one, for sure. It will not be more correct\* than random, but at least more consistent. :)
* I think the whole thing in RESTful interfaces using HTTP errors is a total...
